src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.ts

```diff
@@ -43,4 +43,14 @@ export class HostService {
   getDaemons(hostname: string): Observable<Daemon[]> {
     return this.http.get<Daemon[]>(`${this.baseURL}/${hostname}/daemons`);
   }
+
+  getLabels(): Observable<string[]> {
+    return this.http.get<string[]>('ui-api/host/labels');
+  }
+
+  update(hostname: string, labels: string[]) {
+    let params = new HttpParams();
+    params = params.append('labels', labels.join(','));
+    return this.http.put(`${this.baseURL}/${hostname}`, null, { params: params });
```

Comment:
Thanks for addressing the previous comments.  Just one more thing, putting a JSON body is more common than putting an empty body with URL parameters. I'd suggest putting a JSON body here for consistency with other codes:

```
{
    "labels": ["label1", "label2"]
}
```
The backend code also needs a slight change (make `labels` an optional string list).